src/main/java/com/saucelabs/bamboo/sod/action/BuildConfigurator.java

```diff
@@ -120,8 +120,7 @@ public void println(String x) {
         SauceTunnelManager sauceTunnelManager = SauceConnectFourManagerSingleton.getSauceConnectFourTunnelManager();
         String options = getResolvedOptions(config.getSauceConnectOptions());
         if (config.useGeneratedTunnelIdentifier()) {
-            String tunnelIdentifier = generateTunnelIdentifier(buildContext.getPlanName());
-            customVariableContext.addCustomData(SODKeys.TEMP_TUNNEL_ID, tunnelIdentifier);
+            String tunnelIdentifier = customVariableContext.getVariables(buildContext).get(SODKeys.TUNNEL_ID);
             options = "--tunnel-identifier " + tunnelIdentifier + " " + options;
```

Comment:
I don't think you actually need the custom variable context anymore, you can just do `"--tunnel-identifier ${bamboo.TUNNEL_IDENTIFIER}"` and it'll get evaluated (though maybe getResolvedOptions does the evaluation)